docs/pbc/all/payment-service-provider/202410.0/base-shop/hosted-payment-page-back-button-handling.md

```diff
@@ -0,0 +1,117 @@
+
+    /**
+     * @return array<\SprykerShop\Yves\CheckoutPageExtension\Dependency\Plugin\CheckoutStepPostConditionPluginInterface>
+     */
+    protected function getCheckoutSummaryStepPostConditionPlugins(): array
+    {
+        return [
+            new PaymentAppCancelOrderOnSummaryPageAfterRedirectFromHostedPaymentPagePlugin(),
+        ];
+    }
+}
+```
+
+### 2. State Machine Configuration
+
+Update your OMS state configuration in `SalesPayment/config/Zed/Oms/Subprocess/PaymentCancel01.xml`. Add the `exclude from customer` flag to the "payment cancellation pending" state:
+
+```xml
+<state name="payment cancellation pending" display="oms.state.reservation-cancellation-pending">
+    <flag>exclude from customer</flag>
+</state>
+```
+
+{% info_block warningBox "Important" %}
+
+The `exclude from customer` flag is specifically needed for logged-in customers to prevent cancelled payment orders from appearing in their customer account. Without this flag, logged-in customers would see these cancelled orders in their order history.
+
+For more information about the `exclude from customer` flag, see [Order Process Modelling via State Machines](https://docs.spryker.com/docs/pbc/all/order-management-system/202410.0/base-shop/datapayload-conversion/state-machine/order-process-modelling-via-state-machines.html#state-machine-module).
+
+{% endinfo_block %}
+
+## Testing
+
+1. **Basic Flow Test**
+   - Add products to cart.
+   - Proceed to checkout.
+   - Reach payment step and get redirected to hosted payment page.
+   - Use browser's back button.
+   - Verify that the original order is cancelled.
+   - Verify that a new order can be placed.
+
+2. **Stock Verification Test**
+   - Add last available item of a product to cart.
+   - Proceed to hosted payment page.
+   - Use back button.
+   - Verify that the item becomes available again after order cancellation.
+   - Verify that a new order can be placed with the same item.
+
+3. **Order Status Test**
+   - Log in as a customer.
+   - Place an order that triggers hosted payment page.
+   - Use back button.
+   - Verify order status changes to "payment cancellation pending".
+   - Verify order is not visible in customer account.
+   - Log in to Zed and verify that the order exists with the correct state.
+
+## Troubleshooting
+
+### Order Still Visible in Customer Account
+- Verify `exclude from customer` flag is properly set in state machine configuration.
+- Check if the order state transition to "payment cancellation pending" was successful.
+- Verify the state is properly configured in the OMS.
+
+### Stock Issues
+- Ensure proper order cancellation workflow.
+- Verify stock update triggers.
+
+## Related Developer Guides
+
+* [Payments Feature Overview](/docs/scos/dev/feature-walkthroughs/page.version/payments-feature-walkthrough/payments-feature-walkthrough.html)
+* [State Machines](/docs/scos/dev/back-end-development/data-manipulation/datapayload-conversion/state-machine/state-machine.html)
```

Comment:
Please move this to the related articles section in the front matter for proper rendering. [Example](https://github.com/spryker/spryker-docs/blob/master/docs/pbc/all/cart-and-checkout/202410.0/base-shop/manage-in-the-back-office/define-global-thresholds.md?plain=1#L16)

Reply:
Done